ledger/ledger-api-common/src/main/scala/com/digitalasset/ledger/api/tls/ProtocolDisabler.scala

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.api.tls
+
+import java.security.Security
+
+/** Disables the unwanted legacy SSLv2Hello protocol at the JSSE level.
+  * See: https://www.java.com/en/configure_crypto.html#DisableTLS:~:text=Disable%20TLS%201.0%20and%20TLS%201.1
+  */
+object ProtocolDisabler {
+  val disabledAlgorithmsProperty: String = "jdk.tls.disabledAlgorithms"
+  val sslV2Protocol: String = "SSLv2Hello"
+
+  def disableSSLv2Hello(): Unit =
+    PropertiesUpdater(Security.getProperty, Security.setProperty)
+      .appendToProperty(disabledAlgorithmsProperty, sslV2Protocol)
+}
+
+private[tls] case class PropertiesUpdater(
+    getter: String => String,
+    setter: (String, String) => Unit,
+) {
+  def appendToProperty(name: String, value: String): Unit = {
+    val property = getter(name)
+    val fullVal = value.r.findFirstIn(property).map(_ => property).getOrElse(s"$property, $value")
```

Comment:
Maybe better to split based on comma and then reconstruct string with `mkString`.  The reason is that the value may exist as a substring of another algorithm (e.g. DES and 3DES_EDE_CBC).

Reply:
Fair point!